pkg/virtctl/vmexport/vmexport.go

```diff
@@ -0,0 +1,639 @@
+		return err
+	}
+	defer resp.Body.Close()
+
+	// Check server response
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("Bad status: %s", resp.Status)
+	}
+
+	// Lastly, copy the file to the expected output
+	if err := copyFileWithProgressBar(output, resp); err != nil {
+		return err
+	}
+
+	fmt.Println("Download finished succesfully")
+	return nil
+}
+
+// getUrlFromVirtualMachineExport inspects the VirtualMachineExport status to fetch the extected URL
+func getUrlFromVirtualMachineExport(client kubecli.KubevirtClient, vmexport *exportv1.VirtualMachineExport) (string, string, error) {
+	var downloadUrl string
+	var httpHeader string
+
+	if vmexport.Status.Links.External.Volumes == nil {
+		return "", "", fmt.Errorf("Unable to access the volume info from '%s/%s' VirtualMachineExport", vmexport.Namespace, vmexport.Name)
+	}
+
+	for _, exportVolume := range vmexport.Status.Links.External.Volumes {
+		// Access the requested volume
+		if exportVolume.Name == volumeName {
+			for _, format := range exportVolume.Formats {
+				// If the --raw flag is used, we download the uncompressed file
+				if downloadRaw {
+					if format.Format == exportv1.KubeVirtRaw {
+						downloadUrl = format.Url
+						httpHeader = exportTokenHeader
+						break
+					} else if format.Format == exportv1.Dir {
+						downloadUrl = format.Url
+						httpHeader = dirExportTokenHeader
+						break
+					}
+				} else {
+					if format.Format == exportv1.KubeVirtGz || format.Format == exportv1.ArchiveGz {
+						downloadUrl = format.Url
+						httpHeader = exportTokenHeader
+						break
+					}
+				}
+			}
+		}
+	}
+
+	if downloadUrl == "" {
+		return "", "", fmt.Errorf("Unable to get a valid URL from '%s/%s' VirtualMachineExport", vmexport.Namespace, vmexport.Name)
+	}
+
+	return downloadUrl, httpHeader, nil
+}
+
+// waitForVirtualMachineExport waits for the VirtualMachineExport status to be ready
+func waitForVirtualMachineExport(client kubecli.KubevirtClient, namespace string, interval, timeout time.Duration) error {
+	err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+		vmexport, err := getVirtualMachineExport(client, namespace)
+		if err != nil {
+			return false, err
+		}
+
+		if vmexport == nil || vmexport.Status == nil {
+			return false, nil
+		}
+
+		done := vmexport.Status.Phase == "Ready"
+		if !done {
+			fmt.Printf("Waiting for VM Export %s status to be ready...\n", vmexportName)
+			return false, nil
+		}
+
+		if vmexport.Status.Links == nil || vmexport.Status.Links.External == nil {
+			fmt.Printf("Waiting for VM Export %s external links to be available...\n", vmexportName)
+			return false, nil
+		}
+
+		fmt.Printf("Processing completed successfully\n")
+		return done, nil
+	})
+
+	return err
+}
+
+// handleHTTPRequest generates the GET request with proper certificate handling
+func handleHTTPRequest(client kubecli.KubevirtClient, vmexport *exportv1.VirtualMachineExport, downloadUrl, httpHeader string) (*http.Response, error) {
```

Comment:
Don't think you need to pass httpHeader in, it is always going to be `x-kubevirt-export-token` in all cases.